rainier-core/src/main/scala/com/stripe/rainier/compute/RealOps.scala

```diff
@@ -6,12 +6,30 @@ private[compute] object RealOps {
 
   def unary(original: Real, op: UnaryOp): Real =
     original match {
-      case Infinity    => Infinity
-      case NegInfinity => NegInfinity
+      case Infinity => Infinity
+      case NegInfinity =>
+        op match {
+          case ExpOp => Real.zero
+          case LogOp =>
+            throw new ArithmeticException(
+              "Cannot take the log of a negative number")
+          case AbsOp => Infinity
+        }
+      case Constant(Real.BigZero) =>
+        op match {
+          case ExpOp => Real.one
+          case LogOp => NegInfinity
+          case AbsOp => Real.zero
+        }
       case Constant(value) =>
         op match {
           case ExpOp => Real(Math.exp(value.toDouble))
-          case LogOp => Real(Math.log(value.toDouble))
+          case LogOp =>
+            if (value.toDouble < 0)
+              throw new ArithmeticException(
+                "Cannot take the log of " + value.toDouble)
+            else
+              Real(Math.log(value.toDouble))
           case AbsOp => Real(Math.abs(value.toDouble))
```

Comment:
I'm assuming we have to convert to a double for `LogOp` since seems like there's no built-in way to take the log of a BigDecimal? For `AbsOp` though, could this be `Constant(value.abs)`?

Reply:
Right - nice catch, thank you.